### PR TITLE
Use individual files for changelog entries

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This helps avoid merge conflicts in the CHANGELOG.md file by generating it.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Commit-by-commit.

### CODE REVIEW

File: .github/workflows/changelog.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      - name: 'Update CHANGELOG.md'
+        run: |
+          cmake --build build --target update-changelog
+      - name: 'Push CHANGELOG.md'
+        if: github.event_name == 'push'
+        run: |
+          git add 'CHANGELOG.md'
+          if git diff-index --quiet 'HEAD'; then
+            git commit -m "Update CHANGELOG.md"
+            git puskh
```
Comment (mavam) [inline .github/workflows/changelog.yaml:67]:
```suggestion
            git push
```

File: .github/workflows/changelog.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          ./configure --with-bundled-caf --without-arrow
+      - name: 'Update CHANGELOG.md'
+        run: |
+          cmake --build build --target update-changelog
+      - name: 'Push CHANGELOG.md'
+        if: github.event_name == 'push'
+        run: |
+          git add 'CHANGELOG.md'
+          if git diff-index --quiet 'HEAD'; then
+            git commit -m "Update CHANGELOG.md"
```
Comment (mavam) [inline .github/workflows/changelog.yaml:66]:
```suggestion
            git commit -m "Update CHANGELOG"
```

**Comment by mavam** [comment]
Is this adding a separate commit to master with every merge?

**Comment by dominiklohmann** [comment]
> Is this adding a separate commit to master with every merge?

For every merge that adds a changelog entry, yes. The alternative would be not to have a readable changelog file in the repository that we can link to. That we do regularly, hence the workflow.

**Comment by mavam** [comment]
How about we just update the changelog with every release? It wouldn't change the logic you have in place, but reduce commit churn by quite a bit. 

Or an alternative: amend the merge commit.

**Comment by dominiklohmann** [comment]
> How about we just update the changelog with every release? It wouldn't change the logic you have in place, but reduce commit churn by quite a bit.

I don't see the issue, really. It's one commit per merge immediately after merging a PR, which is less than one a day. It really doesn't add much noise.

We use a pinned VAST version internally that is not tied to our release cycle. If we update on every release only, we lose the ability to just click on the submodule and easily see the most recent changes.

> Or an alternative: amend the merge commit.

Not an option. Let's not get into the business of force-pushing to our default branch.

File: cmake/VASTChangelog.cmake.in [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    "changes"
+    "experimental-features"
+    "features"
+    "bug-fixes")
+  set(titles
+    ":zap: Breaking Changes"
+    ":warning: Changes"
+    ":dna: Experimental Features"
+    ":gift: Features"
+    ":beetle: Bug Fixes")
```
Comment (dominiklohmann) [inline cmake/VASTChangelog.cmake.in:38]: Move to top

File: cmake/VASTChangelog.cmake.in [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  "'changelog/' subdirectory instead.\n"
+  "-->\n\n"
+  "# Changelog\n\n"
+  "This changelog documents all notable changes to @PROJECT_NAME@.\n\n")
+file(GLOB releases LIST_DIRECTORIES ON
+  RELATIVE "${changelog_directory}/" "${changelog_directory}/*")
+list(SORT releases)
+list(REVERSE releases)
+list(PREPEND releases "unreleased")
+list(REMOVE_DUPLICATES releases)
```
Comment (dominiklohmann) [inline cmake/VASTChangelog.cmake.in:18]: Instead of removing duplicates, remove "unreleased" first.

File: cmake/VASTChangelog.cmake.in [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  "This file is generated automatically. Add indivual changelog entries to the\n"
+  "'changelog/' subdirectory instead.\n"
+  "-->\n\n"
+  "# Changelog\n\n"
+  "This changelog documents all notable changes to @PROJECT_NAME@.\n\n")
+file(GLOB releases LIST_DIRECTORIES ON
+  RELATIVE "${changelog_directory}/" "${changelog_directory}/*")
+list(SORT releases)
+list(REVERSE releases)
+list(PREPEND releases "unreleased")
```
Comment (dominiklohmann) [inline cmake/VASTChangelog.cmake.in:17]: Prepend only if we removed "unreleased" before.

File: .github/workflows/changelog.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            echo 'Error: CHANGELOG.md must not be modified in PRs'
+            exit 1
+          fi
+      - name: 'Install Dependencies'
+        run: |
+          apt-get -y install libsnappy-dev libbrotli-dev zlib1g-dev wget \
+            liblz4-dev libzstd-dev ninja-build libpcap-dev libssl-dev \
+            libbenchmark-dev libflatbuffers-dev  flatbuffers-compiler-dev \
+            libyaml-cpp-dev libsimdjson-dev gnupg2 gcc-8 g++-8 python3 \
+            python3-pip python3-venv jq software-properties-common
```
Comment (dominiklohmann) [inline .github/workflows/changelog.yaml:38]: Remove python

Comment (dominiklohmann) [reply .github/workflows/changelog.yaml:38]: And jq

**Review by dominiklohmann — COMMENTED** [review]
Peer-reviewed with @tobim.

**Review by tobim — APPROVED** [review]

**Pull request merged**